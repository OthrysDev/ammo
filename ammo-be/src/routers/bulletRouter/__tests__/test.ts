import app from 'index';
import request from 'supertest';
import {
    bulletMock,
    incorrectBulletMock,
    overweightedBulletMock,
    invalidUrlBulletMock,
} from 'shared/mocks/Bullets';

interface HTTPError {
    error: {
        name: string;
        message: string;
        stack?: string;
        status: number;
        text: string;
        method: string;
        path: string;
    };
}

afterAll(() => {
    app.close();
});

describe('Testing endpoints', () => {
    // FIXME
    // test('Send correct data', async () => {
    // await request(app)
    //     .post('/')
    //     .send({ data: bulletMock })
    //     .expect(200)
    //     .expect((r) => {
    //         const { bullet } = r.body;
    //         expect(bullet).toMatchSnapshot();
    //     });
    // });

    // FIXME
    // it('Send incorrect data - Joi must throw error as method is missing', async () => {
    // await request(app)
    //     .post('/')
    //     .send({ data: incorrectBulletMock })
    //     .expect(400)
    //     .expect((r: HTTPError) => {
    //         expect(r.error.text).toContain('\\"method\\" is required');
    //     });
    // });

    it('Send an overweighted body - Joi must throw error', async () => {
        await request(app)
            .post('/')
            .send({ data: overweightedBulletMock })
            .expect(400)
            .expect((r: HTTPError) => {
                expect(r.error.text).toMatchSnapshot();
            });
    });

    it('Send a fake url - Joi must throw error', async () => {
        await request(app)
            .post('/')
            .send({ data: invalidUrlBulletMock })
            .expect(400)
            .expect((r: HTTPError) => {
                expect(r.error.text).toMatchSnapshot();
            });
    });
});
