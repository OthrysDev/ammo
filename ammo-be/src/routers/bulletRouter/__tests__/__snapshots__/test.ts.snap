// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing endpoints Send a fake url - Joi must throw error 1`] = `"{\\"message\\":\\"\\\\\\"url\\\\\\" with value \\\\\\"AfakeUrlWhosGonnaMiss\\\\\\" fails to match the required pattern: /((([A-Za-z]{3,9}:(?:\\\\\\\\/\\\\\\\\/)?)(?:[-;:&=+$,\\\\\\\\w]+@)?[A-Za-z0-9.-]+|(?:www\\\\\\\\.|[-;:&=+$,\\\\\\\\w]+@)[A-Za-z0-9.-]+)((?:\\\\\\\\/[+~%\\\\\\\\/.\\\\\\\\w\\\\\\\\-_]*)?\\\\\\\\??(?:[-+=&;%@.\\\\\\\\w_]*)#?(?:[.!\\\\\\\\/\\\\\\\\\\\\\\\\\\\\\\\\w]*))?)/\\"}"`;

exports[`Testing endpoints Send an overweighted body - Joi must throw error 1`] = `"{\\"message\\":\\"\\\\\\"request.body\\\\\\" failed custom validation because your string is too heavy, max 50Ko\\"}"`;
